### DISCUSSION TITLE
Setting Redis Key Experation

### DISCUSSION THREAD

**Discussion by jretterer-prol** [mozilla/sccache#1665] [Q&A]
In using sccache with redis along with the allkeys-lru eviction policy I was seeing keys being removed from redis even thought the allocated memory space was not fully utilized. That led me to check the TTL on the redis keys and found one is set for sccache keys.

```
127.0.0.1:6379> TTL c/7/8/c78004b4adb30628728df15ce1e99e9f862814aaa9473c1582b5b996e823439c
(integer) 86258
```

After finding that I tried to find a way to adjust this but I can't seam to find it.  So, is there a way to adjust this expiry time for redis keys or even better, set it so the time is refreshed when keys are read?

**Comment by Xuanwo** [comment]
We currently do not support setting expiration, which is an issue. This has been tracked at https://github.com/mozilla/sccache/issues/1666.

**Reply by eandersson** [reply]
It's probably worth pointing out that there is either no expiration (or a much higher default) in 0.3.3 so anyone that upgrades from 0.3.3 to 0.4.0-pre11 could get significantly worse performance in some scenarios when using Redis.
